Assets/Labrys/FeatureLibrary.cs

```diff
@@ -0,0 +1,195 @@
+            }
+        }
+#endif
+        private FeatureLibrary()
+        {
+            nameToFAID = new Dictionary<string, ushort>();
+            features = new FeatureAsset[0];
+        }
+
+        private bool Add(string name, FeatureAsset feature)
+        {
+            if (name == null || feature == null)
+                return false;
+
+            features[nameToFAID.Count] = feature;
+            nameToFAID.Add(name, (ushort)nameToFAID.Count);
+
+            return true;
+        }
+
+        public FeatureAsset GetFeature(string name) => this[name];
+
+        public FeatureAsset GetFeature(ushort faid) => this[faid];
+
+        public bool Contains(string name) => nameToFAID.ContainsKey(name);
+
+        public bool Contains(ushort faid) => 0 <= faid && faid < features.Length;
+
+        public ushort GetId(string name)
+        {
+            if (nameToFAID.TryGetValue(name, out ushort faid))
+            {
+                return faid;
+            }
+            throw new ArgumentException($"Unknown name: {name}.");
+        }
+
+        public IEnumerator<Entry> GetEnumerator()
+        {
+            if (nameToFAID == null)
+                yield break;
+
+            foreach(KeyValuePair<string, ushort> entry in nameToFAID)
+            {
+                yield return new Entry(entry.Value, entry.Key, this[entry.Value]);
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public object Clone()
+        {
+            FeatureLibrary clone = CreateInstance<FeatureLibrary>();
+
+            clone.TargetDirectory = TargetDirectory;
+
+            clone.features = new FeatureAsset[features.Length];
+            Array.Copy(features, clone.features, features.Length);
+
+            clone.nameToFAID = new Dictionary<string, ushort>();
+            foreach(KeyValuePair<string, ushort> pair in nameToFAID)
+            {
+                clone.nameToFAID.Add(pair.Key, pair.Value);
+            }
+
+            return clone;
+        }
+
+        public override string ToString()
+        {
+            string str = "";
+
+            foreach(Entry e in this)
+            {
+                str += $"{{faid: {e.FaID}, name: {e.Name}, feature: {e.Feature.ToString()}}}\n";
+            }
+
+            return $"TargetDirectory: {TargetDirectory}, Contents: [\n{str}\n]";
+        }
+
+        public struct Entry
+        {
+            public ushort FaID { get; }
+            public string Name { get; }
+            public FeatureAsset Feature { get; }
+
+            public Entry(ushort faid, string name, FeatureAsset feature)
+            {
+                FaID = faid;
+                Name = name;
+                Feature = feature;
+            }
+        }
+
+        #region SERIALIZATION
+        [SerializeField]
+        private List<string> serializedNames;
+        [SerializeField]
+        private List<FeatureAsset> serializedFeatures;
```

Comment:
if i remember right, only the Dictionary has trouble serializing. the `FeatureAsset[] features` should be okay, so based on that, `serializedFeatures` might be unnecessary. If so, it can be replaced with just the list of strings in the same order as the FeatureAsset array.